IdentityCore/src/requests/sdk/MSIDTokenResponseValidator.m

```diff
@@ -207,6 +209,20 @@ - (MSIDTokenResult *)validateAndSaveTokenResponse:(MSIDTokenResponse *)tokenResp
     {
         return nil;
     }
+    
+    //save metadata
+    NSError *updateMetadataError = nil;
+    [metadataCache updateAuthorityMapWithRequestParameters:parameters
+                                            cacheAuthority:tokenResult.accessToken.authority
```

Comment:
don't take authority from access token. this has been changed. take authority from token result instead

Reply:
done